src/api.rs

```diff
@@ -78,36 +82,29 @@ pub struct FeedPlayerResponse {
 #[post("/feed-me", format = "application/json", data = "<payload>")]
 pub fn feed_player(
     payload: JSON<FeedPlayerRequest>,
-    scoreboard: State<Mutex<Scoreboard>>,
+    players: State<PlayerMap>,
     broadcaster: State<HostBroadcaster>,
 ) -> Result<JSON<FeedPlayerResponse>>
 {
     let payload = payload.into_inner();
-    let player_id = payload.player;
+    let id = payload.player;
 
     // Add 1 to the player's score, returning the new score.
-    let score = {
-        let mut scoreboard = scoreboard.lock().expect("Scoreboard mutex was poisoned");
+    let balls = {
+        let mut players = players.write().expect("Players were poisoned");
 
         // Get the player's current score, or return an `InvalidPlayer` error if it's not in
         // the scoreboard.
-        let score = scoreboard
-            .get_mut(&player_id)
-            .ok_or(Error::InvalidPlayer(player_id))?;
-        *score += 1;
-        *score
```

Comment:
Yuppers. Rust is a lot like C/C++ in that most of the fundamental parts of the language are very thin abstractions over how the hardware works. Within an `unsafe` block, you can manipulate raw memory in the same way you would C/C++. Normally, though, Rust restricts you to only performing safe operations, even though you're always working with pointers when you use reference/borrows.

In Rust terms, though, `&usize` is a "reference to a `usize`" or "a `usize` borrow". Under the hood it's still functionally a pointer, hence why you need to manually dereference it, but Rust also "raw pointers", which allow you to do all the unsafe pointer hackery you can do in C/C++.